test/e2e/dra/versionskew.go

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dra
+
+import (
+	"bufio"
+	"context"
+	_ "embed"
+	"fmt"
+	"io"
+	"os/exec"
+	"path"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/version"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/kubernetes/cmd/kubeadm/app/util/errors"
+	"k8s.io/kubernetes/pkg/features"
+	"k8s.io/kubernetes/test/e2e/feature"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2enode "k8s.io/kubernetes/test/e2e/framework/node"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	"k8s.io/kubernetes/test/utils/kindcluster"
+	"k8s.io/kubernetes/test/utils/ktesting"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+//go:embed kind.yaml
+var kindConfig string
+
+// Upgrade/downgrade tests are slow, partly because "make quick-release" below
+// and BuildImage in kindcluster.go are slow.
+//
+// If both were done once, they don't need to be repeated unless something
+// changes. This is not checked, so both steps are always done.
+// Developers who have gone through the full test and want to re-run
+// it more quickly can comment out the invocations.
+// Beware that the apiserver not coming up again can be caused by not
+// having up-to-date release images, so run "make quick-release" manually.
+
+var _ = framework.SIGDescribe("node")(framework.WithLabel("DRA"), feature.DynamicResourceAllocation, framework.WithFeatureGate(features.DynamicResourceAllocation), framework.WithSlow() /* TODO: add feature.KindCommand once jobs are updated*/, func() {
+
+	ginkgo.It("upgrade/downgrade works", func(ctx context.Context) {
```

Comment:
@lauralorenz, @SergeyKanzhelev: once this PR is merged with this initial base test, it would be great to extend it with other scenarios described in https://github.com/kubernetes/kubernetes/issues/128965.